lib/src/util/windows/user.cc

```diff
@@ -0,0 +1,58 @@
+#include <internal/util/windows/user.hpp>
+#include <internal/util/windows/process.hpp>
+#include <internal/util/windows/system_error.hpp>
+#include <internal/util/windows/windows.hpp>
+#include <facter/util/environment.hpp>
+#include <leatherman/logging/logging.hpp>
+
+using namespace std;
+
+namespace facter { namespace util { namespace windows { namespace user {
+
+    bool is_admin()
+    {
+        if (process::supports_elevated_security()) {
+            return process::has_elevated_security();
+        }
+
+        return check_token_membership();
+    }
+
+    bool check_token_membership()
+    {
+        DWORD sid_size = SECURITY_MAX_SID_SIZE;
+        unsigned char sid_buffer[SECURITY_MAX_SID_SIZE];
+        auto sid = static_cast<PSID>(&sid_buffer);
+        if (!CreateWellKnownSid(WinBuiltinAdministratorsSid, nullptr, sid, &sid_size)) {
+            LOG_DEBUG("Failed to create administrators SID: %1%", system_error());
+            return false;
+        }
+
+        if (!IsValidSid(sid)) {
+            LOG_DEBUG("Invalid SID");
+            return false;
+        }
+
+        BOOL is_member;
+        if (!CheckTokenMembership(nullptr, sid, &is_member)) {
+            LOG_DEBUG("Failed to check membership: %1%", system_error());
+            return false;
+        }
+
+        return is_member;
+    }
+
+    string home_dir()
+    {
+        string home, alt;
```

Comment:
Sorry - late to the party on this one.  I'm guessing you went off of the existing Puppet code that I pointed you to... but 

Why not use [GetUserProfileDirectory](https://msdn.microsoft.com/en-us/library/windows/desktop/bb762280%28v=vs.85%29.aspx) here?

Alternatively, you could also use the shell function [SHGetFolderPath](https://msdn.microsoft.com/en-us/library/windows/desktop/bb762181.aspx) with [CSIDL_PROFILE](https://msdn.microsoft.com/en-us/library/windows/desktop/bb762494.aspx) - but looks like there are some caveats.

I'm guessing that we didn't use the API calls when the Windows code was originally written because it was an "easier" approach in Ruby (code wasn't FFI'd at the time for one thing).

I know we *probably* haven't run into issues within Puppet, but I think peeking at a couple of env vars is a fairly naive approach here.